bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -65,144 +63,113 @@ impl S2NHarness {
-                .set_verify_host_callback(HostNameHandler {
-                    expected_server_name: "localhost",
-                })
-                .unwrap(),
-        };
-
-        builder.build().unwrap()
-    }
-
-    /// Set up connections with config and custom IO
-    fn init_conn(&mut self, mode: Mode) -> Result<(), s2n_tls::error::Error> {
-        let client_to_server_ptr = self.client_to_server_buf.get() as *mut c_void;
-        let server_to_client_ptr = self.server_to_client_buf.get() as *mut c_void;
-        let (read_ptr, write_ptr, config, conn) = match mode {
-            Mode::Client => (
-                server_to_client_ptr,
-                client_to_server_ptr,
-                &self.client_config,
-                &mut self.client_conn,
-            ),
-            Mode::Server => (
-                client_to_server_ptr,
-                server_to_client_ptr,
-                &self.server_config,
-                &mut self.server_conn,
-            ),
-        };
-
-        conn.set_blinding(Blinding::SelfService)?
-            .set_config(config.clone())?
-            .set_send_callback(Some(Self::send_cb))?
-            .set_receive_callback(Some(Self::recv_cb))?;
-        unsafe {
-            conn.set_send_context(write_ptr)?
-                .set_receive_context(read_ptr)?;
-        }
-
-        Ok(())
-    }
-
     /// Handshake step for one connection
-    fn handshake_conn(&mut self, mode: Mode) {
+    fn handshake_conn(&mut self, mode: Mode) -> Result<(), Box<dyn Error>> {
         let (conn, handshake_completed) = match mode {
-            Mode::Client => {
-                debug!("Client: ");
-                (&mut self.client_conn, &mut self.client_handshake_completed)
-            }
-            Mode::Server => {
-                debug!("Server: ");
-                (&mut self.server_conn, &mut self.server_handshake_completed)
-            }
+            Mode::Client => (&mut self.client_conn, &mut self.client_handshake_completed),
+            Mode::Server => (&mut self.server_conn, &mut self.server_handshake_completed),
         };
 
         if let Ready(res) = conn.poll_negotiate() {
-            res.unwrap();
+            res?;
             *handshake_completed = true;
         } else {
             *handshake_completed = false;
         }
+        Ok(())
     }
 }
 
 impl TlsBenchHarness for S2NHarness {
-    fn new() -> Self {
-        debug!("----- constructing new s2n-tls harness -----");
-        let mut harness = S2NHarness {
-            client_to_server_buf: Box::pin(UnsafeCell::new(VecDeque::new())),
-            server_to_client_buf: Box::pin(UnsafeCell::new(VecDeque::new())),
-            client_config: Self::create_config(Mode::Client),
-            server_config: Self::create_config(Mode::Server),
-            client_conn: Connection::new_client(),
-            server_conn: Connection::new_server(),
+    fn new(crypto_config: &CryptoConfig) -> Result<Self, Box<dyn Error>> {
```

Comment:
This function is starting to feel a little long, but I'm not necessarily opposed to it 🤷 . Idk, perhaps @maddeleine has thoughts?

Reply:
Yeah, I do prefer the previous version where create_config and create_connection were two separate functions. Any reason why you changed it?

Reply:
I was just playing around with how it looked, since to separate them out into functions you'd need a pretty beefy `match`. I can extract them out again, since it's probably easier to understand for someone reading the code.